cpp/src/gandiva/regex_functions_holder.cc

```diff
@@ -132,12 +131,11 @@ Status LikeHolder::Make(const std::string& sql_pattern,
                   Status::Invalid("Building RE2 pattern '", pcre_pattern,
                                   "' failed with: ", lholder->regex_.error()));
 
-  *holder = std::move(lholder);
-  return Status::OK();
+  return std::move(lholder);
```

Comment:
I think that `std::move()` for local variable in `return` isn't needed.
I think that it'll prevent NRVO (Named Return Value Optimization). See also: https://en.cppreference.com/w/cpp/language/copy_elision#Non-mandatory_copy.2Fmove.28since_C.2B.2B11.29_elision

Reply:
Thanks for pointing it out. I change several usages I can find in the codebase (all in the same file)